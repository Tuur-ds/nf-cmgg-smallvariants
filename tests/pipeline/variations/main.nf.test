nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_variations"

    options "-stub"

    test("pipeline_variations - annotate") {

        when {
            params {
                outdir = "${outputDir}"
                annotate = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("annotate") }
            )
        }

    }

    test("pipeline_variations - annotate + vcfanno") {

        when {
            params {
                outdir = "${outputDir}"
                annotate = true
                vcfanno = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("annotate + vcfanno") }
            )
        }

    }

    test("pipeline_variations - filter") {

        when {
            params {
                outdir = "${outputDir}"
                filter = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("filter") }
            )
        }

    }

    test("pipeline_variations - only_call") {

        when {
            params {
                outdir = "${outputDir}"
                only_call = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("only_call") }
            )
        }

    }

    test("pipeline_variations - only_merge") {

        when {
            params {
                outdir = "${outputDir}"
                only_merge = true
                output_genomicsdb = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("only_merge") }
            )
        }

    }

    test("pipeline_variations - automap") {

        when {
            params {
                outdir = "${outputDir}"
                automap = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("automap") }
            )
        }

    }
}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
