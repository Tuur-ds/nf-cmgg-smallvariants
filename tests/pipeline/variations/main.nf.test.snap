{
    "pipeline_variations - only_merge": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim_haplotypecaller_genomicsdb",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:52:11.995335577"
    },
    "pipeline_variations - filter": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.bed",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.ped",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.html",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:50:35.751317235"
    },
    "pipeline_variations - annotate + vcfanno": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.bed",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.ped",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.html",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:49:26.904699766"
    },
    "pipeline_variations - annotate": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.bed",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.ped",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.html",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:48:06.520399436"
    },
    "pipeline_variations - only_call": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:51:23.297404025"
    },
    "pipeline_variations - automap": {
        "content": [
            [
                "Ashkenazim/NA24143_<version>_<date>/NA24143.bed",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz",
                "Ashkenazim/NA24143_<version>_<date>/NA24143.per-base.bed.gz.csi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.bed",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz",
                "Ashkenazim/NA24149_<version>_<date>/NA24149.per-base.bed.gz.csi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.bed",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.haplotypecaller.g.vcf.gz.tbi",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz",
                "Ashkenazim/NA24385_<version>_<date>/NA24385.per-base.bed.gz.csi",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.bed",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.ped",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz",
                "Ashkenazim/output_<version>_<date>/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample1/sample1.HomRegions.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample1/sample1.HomRegions.pdf",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample1/sample1.HomRegions.strict.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample1/sample1.HomRegions.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample2/sample2.HomRegions.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample2/sample2.HomRegions.pdf",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample2/sample2.HomRegions.strict.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample2/sample2.HomRegions.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample3/sample3.HomRegions.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample3/sample3.HomRegions.pdf",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample3/sample3.HomRegions.strict.cmgg_bio.tsv",
                "Ashkenazim/output_<version>_<date>/automap/haplotypecaller/sample3/sample3.HomRegions.tsv",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "Ashkenazim/qc_<version>_<date>/Ashkenazim.haplotypecaller.html",
                "<version>_<date>/multiqc_report.html",
                "<version>_<date>/samplesheet.csv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-22T13:53:25.411254268"
    }
}